command/sub8_missions/missions/arm_torpedos.py

```diff
@@ -96,18 +122,32 @@ def err():
     @util.cancellableInlineCallbacks
     def fire(self, target):
         self.print_info("FIRING {}".format(target))
+        target_pose = self.targets[target].position
         yield self.sub.move.go(blind=self.BLIND, speed=0.1)  # Station hold
-        target_position = self.targets[target].position
-        yield self.sub.move.depth(-target_position[2]).go(blind=self.BLIND, speed=.1)
-        yield self.sub.move.look_at_without_pitching(target_position).go(
-            blind=self.BLIND, speed=.1)
-        yield self.sub.move.set_position(target_position).backward(1).go(
+        transform = yield self.sub._tf_listener.get_transform('/map', '/base_link')
+        target_position = transform._q_mat.dot(
+                target_pose - transform._p)
+
+        sub_pos = yield self.sub.tx_pose()
+        print('Current Sub Position: ', sub_pos)
+
+        # sub_pos = transform._q_mat.dot(
+                # (sub_pos[0]) - transform._p)
+        # target_position = target_position - sub_pos[0]
+        # yield self.sub.move.look_at_without_pitching(target_position).go(
+            # blind=self.BLIND, speed=.1)
+        print('Map Position: ', target_position)
+        yield self.sub.move.relative(np.array([0, target_position[1], 0])).go(blind=True, speed=.1)
+        yield self.sub.move.relative(np.array([target_position[0], 0, target_position[2]])).go(
             blind=self.BLIND, speed=.1)
+
         self.print_good(
             "{} locked. Firing torpedos. Hit confirmed, good job Commander.".
             format(target))
-        yield self.sub.set('torpedo1', True)
-        yield self.sub.set('torpedo2', True)
+        sub_pos = yield self.sub.tx_pose()
+        print('Current Sub Position: ', sub_pos)
+        yield self.sub.actuators.shoot_torpedo1()
```

Comment:
will need to offset the sub a little bit, since the torpedos aren't exactly in the center of sub